hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieAvroDataBlock.java

```diff
@@ -187,11 +188,12 @@ private RecordIterator(Schema readerSchema, Schema writerSchema, byte[] content)
         this.totalRecords = this.dis.readInt();
       }
 
-      if (recordNeedsRewriteForExtendedAvroTypePromotion(writerSchema, readerSchema)) {
-        this.reader = new GenericDatumReader<>(writerSchema, writerSchema);
+      Schema repairedWriterSchema = AvroSchemaRepair.repairLogicalTypes(writerSchema, readerSchema);
```

Comment:
the writer schema and reader schema are both schema from the log header which is actually the same, the fix seems not working as expected.

Reply:
yeah, looks like a legacy code, from the constructor in line 200, it looks like the same schemas:

```java
    public static RecordIterator getInstance(HoodieAvroDataBlock dataBlock, byte[] content) throws IOException {
      return new RecordIterator(dataBlock.readerSchema, dataBlock.getSchemaFromHeader(), content);
    }
```

And we should invoke `AvroSchemaRepair.repairLogicalTypes(` after checking `recordNeedsRewriteForExtendedAvroTypePromotion` as true because micros to millis check is added there?